perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -5280,6 +5330,19 @@ public void recalculateMemberGroupStatusRecursively(PerunSession sess, Member me
 			return;
 		}
 
+		// if there is union between different VOs, find correct member (from the group's VO)
+		if (member.getVoId() != group.getVoId()) {
+			try {
+				Vo vo = perunBl.getVosManagerBl().getVoById(sess, group.getVoId());
```

Comment:
I guess you could replace this chunk of code with a call of getCorrespondingMembersFromOtherVo(), since you kind of duplicate it in stepIntoGroup() as well. Up to you.

Reply:
Yes, it's a good suggestion, I added it, thanks.